### PR TITLE
Fuzz prep v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/2859

Describe changes: preparation for fuzzing
- Using more `const` keywords
- use logging instead of stderr during config file parsing
- log can use a file set from env variable
- new file init.c to have common initialisation

First commits of #4309 which can be merged independently

Modifies #4239 with needed rebase

### CODE REVIEW

File: qa/coccinelle/banned-functions.cocci [hunk 1]
```diff
@@ -3,7 +3,7 @@ identifier i;
 position p1;
 @@
 
-\(strtok@i\|sprintf@i\|strcat@i\|strcpy@i\|strncpy@i\|strncat@i\|strndup@i\|strchrnul@i\|rand@i\|rand_r@i\|memmem@i\|index@i\|rindex@i\|bzero@i\)(...)@p1
```
Comment (victorjulien) [inline qa/coccinelle/banned-functions.cocci]: memmem was banned as mingw doesn't have it

**Comment by victorjulien** [comment]
Git message for https://github.com/OISF/suricata/pull/4310/commits/3f6821e1ca18f12a5cab983e1c795c1462dd8607 no longer matches actual diff.

**Comment by victorjulien** [comment]
Looking at https://github.com/OISF/suricata/pull/4310/commits/50ad249f5a48ca2e012d5d5ac19fe9d05fcee271 I don't understand what it does, and the git commit message is not helping. Can you expand on that?

File: qa/coccinelle/banned-functions.cocci (hunk 1 above)
Comment (victorjulien) [reply qa/coccinelle/banned-functions.cocci]: I suppose we can have a ```memmem``` implementation for MinGW that just uses our BasicSearch.

Have to then: check for memmem in configure. If it exists: ok. If it doesn't: use wrapper.

**Comment by victorjulien** [comment]
Cherry-picked https://github.com/OISF/suricata/pull/4310/commits/795f4208322a1fea0f2ade342373c07102031b0f, thanks!

The larger init changes are for 6.0dev.

**Comment by catenacyber** [comment]
> Looking at 50ad249 I don't understand what it does

I want to log to a file and specify the file name from an environment variable.
So that I can do this : https://github.com/OISF/suricata/pull/4309/files#diff-0a7aa825d1c585962bdfdee7c85f2213R63

**Pull request closed without merging**